### DISCUSSION TITLE
Tree / TreeNode rendering

### DISCUSSION THREAD

**Discussion by SadE54** [spectreconsole/spectre.console#993] [Q&A]
Hi,

When creating a tree , you're getting a Tree object , the root. When adding a node , you 're getting a TreeNode object
I created a recursive function to display the content of a tree , but it's not possible directly from the root (due to the root is Tree and  all other are TreeNode).
But when you're rendering a Tree , you have 
Tree 
      ---- TreeNode 1
                       ---- TreeNode 2
      ---- TreeNode 3

Could it be possible to choose the tree level start for rendering ? Or at least be able to hide the first level (Tree object)  and rendering only the nodes ?

**Comment by asasine** [comment]
For what it's worth, both `Tree` and `TreeNode` derive from `IHasTreeNodes`, therefore you can write your recursive function using that interface and utilize the extension methods on it such as `AddNode(...)`
